.github/workflows/pkg-pr-new.yml

```diff
@@ -0,0 +1,67 @@
+name: Publish Any Commit
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    services:
+      # Label used to access the service container
+      postgres:
+        # Docker Hub image
+        image: mghayes/multi-postgres:v2
+        ports:
+          - '5455:5432'
+        env:
+          POSTGRES_USER: prisma
+          POSTGRES_PASSWORD: prisma
+          POSTGRES_MULTIPLE_DATABASES: tests,drizzle
+
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: pnpm/action-setup@v4
+        name: Install pnpm
+        id: pnpm-install
+        with:
+          run_install: false
+
+      - name: Use Node.js 22
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          cache: 'pnpm'
+          cache-dependency-path: pnpm-lock.yaml
+
+      - name: Setup turbo cache
+        uses: actions/cache@v3
+        with:
+          path: node_modules/.cache
+          key: turbo-${{ runner.os }}-18-${{ hashFiles('**/pnpm-lock.yaml') }}
+
+      - name: Creating .npmrc
+        run: |
+          cat << EOF > "$HOME/.npmrc"
+            email=michael@hayes.io
+            //registry.npmjs.org/:_authToken=$NPM_TOKEN
+          EOF
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: Install dependencies
+        run: pnpm install --frozen-lockfile
+
+      - name: Build
+        run: pnpm run ci
```

Comment:
This will run tests and linting.  Do you want publishing to be dependent on that? I don't have a specific opinion on this.  I know for my own prs tests are often a blocker for getting things merged, so if the goal is to test things out before it's merged this might not be what you want

Reply:
I've intentionally included the tests to be run but now I think it makes more sense to exclude it 😅